specification/resourcemover/resource-manager/Microsoft.Migrate/preview/2019-10-01-preview/resourcemovercollection.json

```diff
@@ -961,6 +961,55 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/moveCollections/{moveCollectionName}/requiredFor": {
```

Comment:
your api looks like a POST action, something similar you have above:

"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/moveCollections/{moveCollectionName}/commit": {
      "post":

what's the reason to use /requiredFor? can you use:

"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/moveCollections/{moveCollectionName}"

and pass a filter like [here](https://armwiki.azurewebsites.net/rpaas/odata_filtering.html?q=filter)

Reply:
It is not a POST action. It is a GET on the sourceId which is present in query for the given moveCollection.

Basically we ask user to add resources to the moveCollection so that the graph is complete. This API will help user to know that for what other dependency the resource needs to be added.

The GET of moveCollection has the information about the moveCollection. in this case we want the information of the actual ARM resource and not the moveCollection for which the sourceId in the query is required.